build.gradle

```diff
@@ -209,13 +277,50 @@ task opensslTest(type: Test) {
     }
 }
 
+splitTestConfig.each{ testName, filterCfg ->
+    task "${testName}"(type: Test) {
+        include '**/*.class'
+        filter {
+            filterCfg.each{ filter, values ->
+                values.each{ value ->
+                    "${filter}" "${value}"
+                }
+            }
+        }
+        maxParallelForks = 8
+        jvmArgs += "-Xmx3072m"
+        if (JavaVersion.current() > JavaVersion.VERSION_1_8) {
+            jvmArgs += "--add-opens=java.base/java.io=ALL-UNNAMED"
+        }
+        retry {
+            failOnPassedAfterRetry = false
+            maxRetries = 5
+        }
+        jacoco {
+            excludes = [
+                "com.sun.jndi.dns.*",
+                "com.sun.security.sasl.gsskerb.*",
+                "java.sql.*",
+                "javax.script.*",
+                "org.jcp.xml.dsig.internal.dom.*",
+                "sun.nio.cs.ext.*",
+                "sun.security.ec.*",
+                "sun.security.jgss.*",
+                "sun.security.pkcs11.*",
+                "sun.security.smartcardio.*",
+                "sun.util.resources.provider.*"
+            ]
+        }
+    }
+}
+
 task copyExtraTestResources(dependsOn: testClasses) {
     copy {
         from 'src/test/resources'
         into 'build/testrun/test/src/test/resources'
     }
 }
-tasks.test.dependsOn(copyExtraTestResources, opensslTest)
+tasks.test.dependsOn(copyExtraTestResources, opensslTest, taskNames)
```

Comment:
Could you please exclude  `opensslTest` here since you already defined the `sslTest`  task for it?

Reply:
Thanks for noticing that. I have to dig deeper in how `**/OpenSSL*.class` correlates to test names. Split on `org.opensearch.security.ssl.OpenSSL*` has moved some of the long running tests to `sslTest` job but not all of them.